src/groups/mqb/mqbstat/mqbstat_queuestats.cpp

```diff
@@ -100,6 +100,39 @@ bool filterDirect(const bmqst::TableRecords::Record& record)
     return record.type() == bmqst::StatContext::e_TOTAL_VALUE;
 }
 
+/// Helper method to calculate queue utilization (messages/bytes).
+bsls::Types::Int64
+queueUtilization(const bmqst::StatContext&                 context,
+                 const bmqst::StatValue::SnapshotLocation& latestSnapshot,
+                 const bmqst::StatValue::SnapshotLocation& oldestSnapshot,
+                 DomainQueueStats::Enum                    currentValue,
+                 DomainQueueStats::Enum                    limitValue)
+{
+    double avg = 0;
+    if (latestSnapshot > oldestSnapshot) {
+        avg = bmqst::StatUtil::average(
+            context.value(bmqst::StatContext::e_DIRECT_VALUE, currentValue),
+            latestSnapshot,
+            oldestSnapshot);
+    }
+    else {
+        avg = static_cast<double>(bmqst::StatUtil::value(
+            context.value(bmqst::StatContext::e_DIRECT_VALUE, currentValue),
+            latestSnapshot));
+    }
```

Comment:
What is the exact semantic of the calculated value? Is it "current utilization", "max utilization", etc?

Reply:
I consider that utilization is calculated as average value of current messages/bytes within publish period divided by limit value. Snapshot interval is usually 1 sec, publish interval is usually 10 sec. So, the average value of messages/bytes for the last 10 snapshots is calculated. Added comments into the code describing this logic.

Reply:
I think we should keep the semantics aligned with the existing history metrics, which are either current or max. Given the choice between the two, I think max is the more useful option. Domains already provide the ability to alarm (internally in the broker) if capacity crosses a threshold (e.g. 80% full), so exposing that max would be the the closest stat to the stat triggering that alarm.

Reply:
Agree, refactored to use max value for calculation of queue utilization.